src/redux/user/user.sagas.js

```diff
@@ -75,6 +76,38 @@ export const loginUser = (data) => {
   return setItems(query, data);
 };
 
+export function* handleGoogleUserLogin({ payload }) {
+  try {
+    yield put(setUserLoading(true));
+    const user = yield call(
+      getItems,
+      `
+    mutation($id_token:String!){googleUser(id_token:$id_token){
+      firstName,
+      lastName,
+      email,
+      credentials{
+        source,
+        tokenPass
+      }
+      token
+} 
+
+}
+  `,
+      {
+        id_token: payload.id_token
+      }
+    );
+    yield put(setUser(user.data.googleUser));
+    yield setToLocalStorage('accessToken', user.data.googleUser.token);
+    yield put(setUserLoading(false));
+    yield put(push('/profile'));
+  } catch (error) {
+    yield put(setUserError(error.message.replace('GraphQL error: ', '')));
+  }
```

Comment:
It is a good practice to have a `final` cell in redux, to stop `loading`, please add it.

Reply:
Fixed